src/main/webapp/assets/js/signUpScript.js

```diff
@@ -52,23 +69,195 @@ async function onLoad() {
   collegeDataList.appendChild(fragment);
   collegeDataList.setAttribute('id', 'colleges');
 
-  // When users select an option from the dropdown, set the college id.
-  document.getElementById('colleges-input-form').addEventListener('change', setCollegeID);
+  // When users submit the form, initiate form validation.
+  emailForm = document.getElementById('email-form');
+  submitFormButton = document.getElementById('form-submit');
+  subscribeFormSubmitted = document.getElementById('subscribe-form-submitted');
+  unsubscribeFormSubmitted = document.getElementById('unsubscribe-form-submitted');
+  submitFormButton.addEventListener('click', submitForm);
+}
+
+/**
+ * On click of the submit button, validate form data and send data to the servlet.
+ * @return {void}
+ */
+async function submitForm() {
+  // Disable multiple submissions.
+  submitFormButton.disabled = true;
+
+  if (validateForm()) {
+    emailForm.action = '/user';
+
+    // Based on user subscribing or unsubscribing, show confirmation that form
+    // was submitted.
+    if (document.getElementById('subscribe').checked) {
+      subscribeFormSubmitted.style.display = 'block';
+    } else {
+      unsubscribeFormSubmitted.style.display = 'block';
```

Comment:
So here you could set the subtitle text instead of showing the different ones. This might help with redundant code/objects.